### PR TITLE
End of Sprint documentation PR

### PR DESCRIPTION
**Pull request by chibenwa**
Currently contains documentation changes for:
 - Explaining a bit better **MailRepository** feature
 - JAMES-2293 WebAdmin routes for MailRepository management
 - JAMES-2294 WebAdmin routes for mail reprocessing
 - JAMES-2291 CassandraMailRepository

Additionally a bit of configuration is modified in order to use directly `cassandra://...` as `ToRepository` paramenters.

### CODE REVIEW

**Comment by chibenwa** [comment]
Aborted to not occupy Jenkins runners.

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[08513fdb839359a72889d9e3fdecfc10028e299b] delayShouldAtLeastBeTheOneSpecified()  Time elapsed: 1.009 sec  <<< FAILURE!
[08513fdb839359a72889d9e3fdecfc10028e299b] java.lang.AssertionError: 
[08513fdb839359a72889d9e3fdecfc10028e299b] 
[08513fdb839359a72889d9e3fdecfc10028e299b] Expecting:
[08513fdb839359a72889d9e3fdecfc10028e299b]  <1000L>
[08513fdb839359a72889d9e3fdecfc10028e299b] to be greater than:
[08513fdb839359a72889d9e3fdecfc10028e299b]  <1000L> 
[08513fdb839359a72889d9e3fdecfc10028e299b] 	at org.apache.james.queue.api.DelayedMailQueueContract.delayShouldAtLeastBeTheOneSpecified(DelayedMailQueueContract.java:115)
[08513fdb839359a72889d9e3fdecfc10028e299b] 
[08513fdb839359a72889d9e3fdecfc10028e299b] 
[08513fdb839359a72889d9e3fdecfc10028e299b] Results :
[08513fdb839359a72889d9e3fdecfc10028e299b] 
[08513fdb839359a72889d9e3fdecfc10028e299b] Failed tests: 
[08513fdb839359a72889d9e3fdecfc10028e299b]   DelayedMailQueueContract.delayShouldAtLeastBeTheOneSpecified:115 
[08513fdb839359a72889d9e3fdecfc10028e299b] Expecting:
[08513fdb839359a72889d9e3fdecfc10028e299b]  <1000L>
[08513fdb839359a72889d9e3fdecfc10028e299b] to be greater than:
[08513fdb839359a72889d9e3fdecfc10028e299b]  <1000L> 
[08513fdb839359a72889d9e3fdecfc10028e299b] 
[08513fdb839359a72889d9e3fdecfc10028e299b] Tests run: 69, Failures: 1, Errors: 0, Skipped: 0
```

test this please

File: src/site/markdown/server/manage-webadmin.md [hunk 1]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+curl -XGET http://ip:port/mailRepositories
+```
+
+The answer looks like:
+
+```
+[
+    {
+        "repository": "file://var/mail/error/",
+        "encodedUrl": "file%3A%2F%2Fvar%2Fmail%2Ferror%2F"
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: id x4

File: src/site/markdown/server/manage-webadmin.md [hunk 2]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+        "encodedUrl": "file%3A%2F%2Fvar%2Fmail%2Fspam%2F"
+    },
+    {
+        "repository": "file://var/mail/address-error/",
+        "encodedUrl": "file%3A%2F%2Fvar%2Fmail%2Faddress-error%2F"
+    }
+]
+```
+
+You can use `encodedUrl` to access the repository.
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: id

File: src/site/markdown/server/manage-webadmin.md [hunk 3]
```diff
@@ -1,7 +1,8 @@
 Web administration for JAMES
 ============================
 
-The web administration supports for now the CRUD operations on the domains,the users, their mailboxes and their quotas, as described in the following sections.
+The web administration supports for now the CRUD operations on the domains,the users, their mailboxes and their quotas,
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: missing space after comma

File: src/site/markdown/server/manage-webadmin.md [hunk 4]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+ - 200: The list of mail repositories
+ - 500: Internal error
+
+### Getting additional information for a mail repository
+
+```
+curl -XGET http://ip:port/mailRepositories/encodedUrlOfTheRepository/
+```
+
+Resource name `encodedUrlOfTheRepository` should be the encoded URL of an existing mail repository. Example:
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: (given by the id field)

File: src/site/markdown/server/manage-webadmin.md [hunk 5]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+```
+curl -XGET http://ip:port/mailRepositories/file%3A%2F%2Fvar%2Fmail%2Ferror%2F/
+```
+
+The answer looks like:
+
+```
+{
+   "repository": "file://var/mail/error/",
+   "encodedUrl": "file%3A%2F%2Fvar%2Fmail%2Ferror%2F",
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: id

File: src/site/markdown/server/manage-webadmin.md [hunk 6]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+### Listing mails contained in a mail repository
+
+```
+curl -XGET http://ip:port/mailRepositories/encodedUrlOfTheRepository/mails
+```
+
+Resource name `encodedUrlOfTheRepository` should be the encoded URL of an existing mail repository. Example:
+
+```
+curl -XGET http://ip:port/mailRepositories/file%3A%2F%2Fvar%2Fmail%2Ferror%2F/
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: missing /mails

File: src/site/markdown/server/manage-webadmin.md [hunk 7]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+Note that this can be used to read mail details.
+
+You can pass additional URL parameters to this call in order to limit the output:
+ - A limit: no more elements than the specified limit will be returned. This needs to be strictly positive. If no value is specified, no limit will be applied.
+ - An offset: allow to skip elements. This needs to be positive. Default value is zero.
+
+Example:
+
+```
+curl -XGET http://ip:port/mailRepositories/file%3A%2F%2Fvar%2Fmail%2Ferror%2F/?limit=100&offset=500
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: missing mails resource

File: src/site/markdown/server/manage-webadmin.md [hunk 8]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+
+Resource name `encodedUrlOfTheRepository` should be the encoded URL of an existing mail repository. Resource name `mailKey` should be the key of a mail stored in that repository. Example:
+
+```
+curl -XDELETE http://ip:port/mailRepositories/file%3A%2F%2Fvar%2Fmail%2Ferror%2F/mails/mail-key-1
+```
+
+Response codes:
+
+ - 204: This mail no longer exist in this repository
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: exists

File: src/site/markdown/server/manage-webadmin.md [hunk 9]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+Location: /tasks/3294a976-ce63-491e-bd52-1b6f465ed7a2
+```
+
+Response codes:
+
+ - 201: Success. Corresponding task id is returned.
+ - 404: Could not find that mail repository
+ - 500: Internal error
+
+The scheduled task will have the following type `reprocessingAllTask` and the following `additionalInformation`:
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: reprocessingAllMails?

File: src/site/markdown/server/manage-webadmin.md [hunk 10]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+  "targetQueue":"spool",
+  "targetProcessor":"transport",
+  "initialCount": 243,
+  "remainingCount": 17
+}
+```
+
+### Reprocessing a specific mail from a mail repository
+
+To reprocess mails from a specific mail from a mail repository:
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: s/mails from //

File: src/site/markdown/server/manage-webadmin.md [hunk 11]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+Location: /tasks/3294a976-ce63-491e-bd52-1b6f465ed7a2
+```
+
+Response codes:
+
+ - 201: Success. Corresponding task id is returned.
+ - 404: Could not find that mail repository
+ - 500: Internal error
+
+The scheduled task will have the following type `reprocessingOneTask` and the following `additionalInformation`:
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: s/Task/Mail/?

File: src/homepage/_posts/2018-01-26-admin-features.markdown [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+A brand new webadmin API allows you to handle the content of these repositores, see the documentation in [manage-webadmin].
+It allows you to list the repositories, their content, but also to reprocess one or several mails in the processor and queue
+of your choice.
+
+Oh! And bonus, now you can forget this non scalable file repository and use the brand new Cassandra repository!
+
+On the same documentation page you will also find the new mail queue management API, done to list queued mails, remove some
+of them regarding different criteria, but also flush delayed mails.
+
```
Comment (chibenwa) [inline src/homepage/_posts/2018-01-26-admin-features.markdown]: Most of the work on these already usable features is done, but we still have some ideas for related improvements. [Contributions] are welcomed!

[Contributions]: https://issues.apache.org/jira/browse/JAMES-2321?jql=project%20%3D%20JAMES%20AND%20status%20%3D%20Open%20AND%20%20component%20%20in(%22MailStore%20%26%20MailRepository%22%20%2C%20Queue)%20AND%20labels%20%3D%20feature%20AND%20type%20%20not%20in%20(Bug)%20

**Comment by chibenwa** [comment]
RCPTs: server-user / server-dev
Subject: Feature announcement - call to contribution

```
We are really happy to announce new management features on top of WebAdmin. They include:

 - MailRepositories management (list, read, delete, reprocess mails)
 - MailQueue management (browse, read mails, remove mails, flush)
 - A distributed implementation for MailRepositories on top of Cassandra

These changes are further described in this [blog post] and will be included in an incoming 3.1.0 release.

Any help is welcome in order to support the ongoing development effort toward 3.1.0 release. Here are some contribution ideas:

 - [JAMES-2319] As an administrator I want to clear a mailQueue
 - [JAMES-2320] MailQueue element removal should be backed by a Task
 - [JAMES-2321] WebAdmin should allow to see more details about Mails
 - [JAMES-2314] Better handle exceptions in webAdmin
 - [JAMES-2301] JMS mail queue do not preserve per-recipient headers
 - [JAMES-2309] Long overflow in JMS delays

Of course other ideas are more than welcome, and we can provide support on GitHub as well as our [gitter chat].

[blog post]: http://james.apache.org/james/update/2018/01/26/admin-features.html
[JAMES-2319]: https://issues.apache.org/jira/browse/JAMES-2319
[JAMES-2320]: https://issues.apache.org/jira/browse/JAMES-2320
[JAMES-2321]: https://issues.apache.org/jira/browse/JAMES-2321
[JAMES-2314]: https://issues.apache.org/jira/browse/JAMES-2314
[JAMES-2301]: https://issues.apache.org/jira/browse/JAMES-2301
[JAMES-2309]: https://issues.apache.org/jira/browse/JAMES-2309
[gitter chat]: https://gitter.im/apache/james-project
```

**Comment by chibenwa** [comment]
Tweet proposal:

```
@ApacheJames now supports mailRepository and mailQueue management through webAdmin.

Read more and help us impove these features: tinyUrl(@rouazana blog post)

[Funny GIF showing that we are cool and happy]
```

**Comment by chibenwa** [comment]
We should also plan a similar annoucment on Gitter once the website is deployed.

File: src/site/markdown/server/manage-webadmin.md [hunk 13]
```diff
@@ -591,6 +592,329 @@ Response codes:
  - 400: Group structure or member is not valid
  - 500: Internal error
 
+## Administrating mail repositories
```
Comment (mbaechler) [inline src/site/markdown/server/manage-webadmin.md]: s/Administrating/Operating ?

File: src/site/markdown/server/manage-webadmin.md (hunk 4 above)
Comment (mbaechler) [reply src/site/markdown/server/manage-webadmin.md]: it's not the resource name but it's id then

File: src/site/markdown/server/manage-webadmin.md [hunk 14]
```diff
@@ -591,6 +592,329 @@ Response codes:
...
+ - 404: This repository can not be found
+ - 500: Internal error
+
+### Listing mails contained in a mail repository
+
+```
+curl -XGET http://ip:port/mailRepositories/encodedUrlOfTheRepository/mails
+```
+
+Resource name `encodedUrlOfTheRepository` should be the encoded URL of an existing mail repository. Example:
```
Comment (mbaechler) [inline src/site/markdown/server/manage-webadmin.md]: Resource id

File: src/site/markdown/server/manage-webadmin.md (hunk 13 above)
Comment (chibenwa) [reply src/site/markdown/server/manage-webadmin.md]: I don't really see the difference between both terms. Especially "Administrating" is used throughout the **WebAdmin** documentation page.

Any rational for such changes?

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**